plugins/modules/cloud/opennebula/one_template.py

```diff
@@ -0,0 +1,265 @@
+        IO = "threads",
+        SIZE = "" ]
+      MEMORY = "2048"
+      NIC = [
+        MODEL = "virtio",
+        NETWORK = "testnet",
+        NETWORK_UNAME = "oneadmin" ]
+      OS = [
+        ARCH = "x86_64",
+        BOOT = "disk0" ]
+      SCHED_REQUIREMENTS = "CLUSTER_ID=\\"100\\""
+      VCPU = "2"
+
+- name: Delete the TEMPLATE by id
+  community.general.one_template:
+    id: 6459
+    state: absent
+'''
+
+RETURN = '''
+id:
+    description: template id
+    type: int
+    returned: when C(state=present)
+    sample: 153
+name:
+    description: template name
+    type: str
+    returned: when C(state=present)
+    sample: app1
+template:
+    description: the parsed template
+    type: dict
+    returned: when C(state=present)
+group_id:
+    description: template's group id
+    type: int
+    returned: when C(state=present)
+    sample: 1
+group_name:
+    description: template's group name
+    type: str
+    returned: when C(state=present)
+    sample: one-users
+owner_id:
+    description: template's owner id
+    type: int
+    returned: when C(state=present)
+    sample: 143
+owner_name:
+    description: template's owner name
+    type: str
+    returned: when C(state=present)
+    sample: ansible-test
+'''
+
+
+from ansible_collections.community.general.plugins.module_utils.opennebula import OpenNebulaModule
+
+
+class TemplateModule(OpenNebulaModule):
+    def __init__(self):
+        argument_spec = dict(
+            id=dict(type='int', required=False),
+            name=dict(type='str', required=False),
+            state=dict(type='str', choices=['present', 'absent'], default='present'),
+            template=dict(type='str', required=False),
+        )
+
+        mutually_exclusive = [
+            ['id', 'name']
+        ]
+
+        required_one_of = [('id', 'name')]
+
+        OpenNebulaModule.__init__(self, argument_spec, supports_check_mode=True, mutually_exclusive=mutually_exclusive, required_one_of=required_one_of)
+
+    def run(self, one, module, result):
+        params = module.params
+        id = params.get('id')
+        name = params.get('name')
+        desired_state = params.get('state')
+        template_data = params.get('template')
+
+        self.result = {}
+
+        template = self.get_template_instance(id, name)
+        needs_creation = False
+        if not template and desired_state != 'absent':
+            if id:
+                module.fail_json(msg="There is no template with id=" + str(id))
+            else:
+                needs_creation = True
+
+        if desired_state == 'absent':
+            self.result = self.delete_template(template)
+        else:
+            if needs_creation:
+                if not template_data:
+                    module.fail_json(msg="To create a new template, the `template` attribute needs to be set.")
+
+                self.result = self.create_template(name, template_data)
+            else:
+                if template_data:
```

Comment:
This `if` would not be needed.

Reply:
I fail to understand why this `if` would not be needed with `required_if` in place...  did you mean the `if not template_data:` one above that?

Reply:
If `state` is `present`, `template_data` will always be there, so neither this `if` nor the one above is needed.

(Of course the user can provide an empty template, but that is probably not accepted by OpenNebula?)

Reply:
ah, I get you now.  My idea here was that this module can also be used to fetch an existing template's information without modifying it.

Reply:
thinking about it I think an inventory module or something would be more appropriate for that use case if needed.  I'll get rid of this in this PR then.

Reply:
Better add an _info module for retrieving templates. Regular modules should modify things, not retrieve things.